ambry-protocol/src/main/java/com/github/ambry/protocol/FileCopyGetMetaDataResponse.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class FileCopyGetMetaDataResponse extends Response {
+  private final int numberOfLogfiles;
+  private final List<LogInfo> logInfos;
+  private static final short File_Copy_Protocol_Metadata_Response_Version_V1 = 1;
+
+  public FileCopyGetMetaDataResponse(short versionId, int correlationId, String clientId, int numberOfLogfiles,
+      List<LogInfo> logInfos, ServerErrorCode errorCode) {
+    super(RequestOrResponseType.FileCopyGetMetaDataResponse, versionId, correlationId, clientId, errorCode);
+    validateVersion(versionId);
+    this.numberOfLogfiles = numberOfLogfiles;
+    this.logInfos = logInfos;
+  }
+
+  public static FileCopyGetMetaDataResponse readFrom(DataInputStream stream) throws IOException {
+    RequestOrResponseType type = RequestOrResponseType.values()[stream.readShort()];
+    if (type != RequestOrResponseType.FileCopyGetMetaDataResponse) {
+      throw new IllegalArgumentException("The type of request response is not compatible");
+    }
+    short versionId = stream.readShort();
+    int correlationId = stream.readInt();
+    String clientId = Utils.readIntString(stream);
+    ServerErrorCode errorCode = ServerErrorCode.values()[stream.readShort()];
+
+    if(errorCode != ServerErrorCode.No_Error) {
+      return new FileCopyGetMetaDataResponse(versionId, correlationId, clientId, -1, new ArrayList<>(), errorCode);
```

Comment:
Why is numberOfLogFiles being set as -1 here? Do add explanation

Reply:
Set to 0 and added explanation.